arch/x86/apic.c

```diff
@@ -155,3 +159,53 @@ void init_apic(unsigned int cpu, apic_mode_t mode) {
     spiv.apic_enable = 1;
     apic_write(APIC_SPIV, spiv.reg);
 }
+
+void init_apic_timer(void) {
+    if (apic_get_mode() < APIC_MODE_XAPIC) {
+        return;
+    }
+
+    apic_lvt_timer_t timer;
+    printk("Initializing local APIC timer\n");
+
+    /* Set the counter to the max value (0xFFFFFFFF) */
+    apic_write(APIC_TMR_DCR, APIC_TIMER_DIVIDE_BY_16);
+    apic_write(APIC_TMR_ICR, _U32(-1));
+
+    /* Sleep for 100ms to calibrate, count the ticks */
+    pit_sleep(100);
+
+    /* One shot mode to see how many ticks over 100ms */
+    timer.timer_mode = APIC_LVT_TIMER_ONE_SHOT;
+    apic_write(APIC_LVT_TIMER, timer.reg);
+
+    /* Calibrate */
```

Comment:
ideally run the whole calibration in a loop, finding the minimal amount of ticks

Reply:
Updated. I wouldn't spend more than 1s in total for the setup not to delay the startup